src/main/java/org/opentripplanner/middleware/controllers/api/BugsnagController.java

```diff
@@ -0,0 +1,44 @@
+package org.opentripplanner.middleware.controllers.api;
+
+import com.mongodb.client.model.Filters;
+import org.bson.conversions.Bson;
+import org.opentripplanner.middleware.bugsnag.EventSummary;
+import org.opentripplanner.middleware.models.BugsnagEvent;
+import org.opentripplanner.middleware.models.BugsnagProject;
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.persistence.TypedPersistence;
+import org.opentripplanner.middleware.utils.JsonUtils;
+import spark.Request;
+import spark.Response;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Responsible for providing the current set of Bugsnag events to the calling service
+ */
+public class BugsnagController {
+
+    private static TypedPersistence<BugsnagEvent> bugsnagEvents = Persistence.bugsnagEvents;
+    private static TypedPersistence<BugsnagProject> bugsnagProjects = Persistence.bugsnagProjects;
+
+    /**
+     * Get all Bugsnag events from Mongo and replace the project id with the project name and return
+     */
+    public static String getEventSummary(Request request, Response response) {
+        response.type("application/json");
+
+        List<EventSummary> eventSummaries = new ArrayList<>();
+        List<BugsnagEvent> events = bugsnagEvents.getAll();
+
+        // FIXME: Group by error/project type?
+        for (BugsnagEvent event : events) {
+            Bson filter = Filters.eq("projectId", event.projectId);
+            BugsnagProject project = bugsnagProjects.getOneFiltered(filter);
+            eventSummaries.add(new EventSummary(project, event));
```

Comment:
Since it could take up to a minute for a new project to get inserted into the database given the schedule of the BugsnagProjectJob, this should still have some kind of check to make sure the project is not null.

Reply:
@evansiroky - There is a null check in EventSummary. I can move it here if you prefer, but it seems cleaner where it is. I've also updated the default values for project id and name within EventSummary so it is clear if these aren't available.

Reply:
Ah, I see that now. Good thinking with the default values.